frontend/ui/network/wpa_supplicant.lua

```diff
@@ -73,14 +73,24 @@ function WpaSupplicant:authenticateNetwork(network)
         wcli:removeNetwork(nw_id)
         return false, _("An error occurred while selecting network.")
     end
-    if not network.psk then
-        network.psk = calculatePsk(network.ssid, network.password)
-        self:saveNetwork(network)
-    end
-    re = wcli:setNetwork(nw_id, "psk", network.psk)
-    if re == 'FAIL' then
-        wcli:removeNetwork(nw_id)
-        return false, _("An error occurred while setting password.")
+    -- if password is empty it’s an open AP
+    if network.password and (network.password == nil or string.len(network.password) == 0) then -- Open AP
+        re = wcli:setNetwork(nw_id, "key_mgmt", "NONE")
+        if re == 'FAIL' then
```

Comment:
I know it already said so before, but let's keep it consistent.
```suggestion
        if re == "FAIL" then
```

Reply:
They are other simple quote in wpa_supplicant.lua, do you want that I make them all double?

Reply:
Sure, if you like.